find-current-git-tag.sh: warning: Could not find remote branch ... to clone / fatal: Remote branch ... not found in upstream origin

In https://github.com/www-learn-study/saraswati.learn.study/actions/runs/6729181929/job/18289629833?pr=20 I ran into this:

```
Run rossjrw/pr-preview-action@183082fd714654433c8e2f6daedbfb4f20f2a94a
  with:
    source-dir: BUILT/site/
    deploy-repository: www-learn-study/previews
    preview-branch: gh-pages
    umbrella-dir: pr-preview
    action: auto
Run echo "action=$action" >> $GITHUB_ENV
  echo "action=$action" >> $GITHUB_ENV
  echo "targetdir=$umbrella/pr-$pr" >> $GITHUB_ENV
  echo "pr=$pr" >> $GITHUB_ENV
  
  org=$(echo "$deployrepo" | cut -d "/" -f 1)
  thirdleveldomain=$(echo "$deployrepo" | cut -d "/" -f [2](https://github.com/www-learn-study/saraswati.learn.study/actions/runs/6729181929/job/18289629833?pr=20#step:5:2))
  
  if [ ! -z "$customurl" ]; then
    pagesurl="$customurl"
  elif [ "${org}.github.io" == "$thirdleveldomain" ]; then
    pagesurl="${org}.github.io"
  else
    pagesurl=$(echo "$deployrepo" | sed 's/\//.github.io\//')
  fi
  
  echo "pagesurl=$pagesurl" >> $GITHUB_ENV
  
  echo "emptydir=$(mktemp -d)" >> $GITHUB_ENV
  echo "datetime=$(date '+%Y-%m-%d %H:%M %Z')" >> $GITHUB_ENV
  
  echo "actionref=$actionref" >> $GITHUB_ENV
  echo "actionrepo=$actionrepo" >> $GITHUB_ENV
  echo "deployrepo=$deployrepo" >> $GITHUB_ENV
  echo "token=$token" >> $GITHUB_ENV
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    action: auto
    umbrella: pr-preview
    pr: 20
    actionref: 18[3](https://github.com/www-learn-study/saraswati.learn.study/actions/runs/6729181929/job/18289629833?pr=20#step:5:3)082fd71[4](https://github.com/www-learn-study/saraswati.learn.study/actions/runs/6729181929/job/18289629833?pr=20#step:5:4)6[5](https://github.com/www-learn-study/saraswati.learn.study/actions/runs/6729181929/job/18289629833?pr=20#step:5:5)4433c8e2f[6](https://github.com/www-learn-study/saraswati.learn.study/actions/runs/6729181929/job/18289629833?pr=20#step:5:6)daedbfb4f20f2a94a
    actionrepo: rossjrw/pr-preview-action
    customurl: 
    deployrepo: www-learn-study/previews
    token: 
Run action_version=$("$GITHUB_ACTION_PATH/lib/find-current-git-tag.sh" -p $actionrepo -f $actionref)
  action_version=$("$GITHUB_ACTION_PATH/lib/find-current-git-tag.sh" -p $actionrepo -f $actionref)
  echo "action_version=$action_version" >> "$GITHUB_ENV"
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    action: auto
    targetdir: pr-preview/pr-20
    pr: 20
    pagesurl: www-learn-study.github.io/previews
    emptydir: /tmp/tmp.bCj8nD[7](https://github.com/www-learn-study/saraswati.learn.study/actions/runs/6729181929/job/18289629833?pr=20#step:5:7)DOE
    datetime: 2023-11-02 06:39 UTC
    actionref: 1[8](https://github.com/www-learn-study/saraswati.learn.study/actions/runs/6729181929/job/18289629833?pr=20#step:5:9)3082fd714654433c8e2f6daedbfb4f20f2a[9](https://github.com/www-learn-study/saraswati.learn.study/actions/runs/6729181929/job/18289629833?pr=20#step:5:10)4a
    actionrepo: rossjrw/pr-preview-action
    deployrepo: www-learn-study/previews
    token: 
Determining Git tag for rossjrw/pr-preview-action/183082fd7[14](https://github.com/www-learn-study/saraswati.learn.study/actions/runs/6729181929/job/18289629833?pr=20#step:5:15)654433c8e2f6daedbfb4f20f2a94a
Cloning repository rossjrw/pr-preview-action at ref [18](https://github.com/www-learn-study/saraswati.learn.study/actions/runs/6729181929/job/18289629833?pr=20#step:5:19)3082fd714654433c8e2f6daedbfb4f[20](https://github.com/www-learn-study/saraswati.learn.study/actions/runs/6729181929/job/18289629833?pr=20#step:5:21)f2a94a
Cloning into bare repository 'bare_pr_preview'...
warning: Could not find remote branch 183082fd714654433c8e2f6daedbfb4f20f2a94a to clone.
fatal: Remote branch 183082fd714654433c8e2f6daedbfb4f20f2a94a not found in upstream origin
/home/runner/work/_actions/rossjrw/pr-preview-action/183082fd714654433c8e2f6daedbfb4f20f2a94a/lib/find-current-git-tag.sh: line [29](https://github.com/www-learn-study/saraswati.learn.study/actions/runs/6729181929/job/18289629833?pr=20#step:5:30): cd: bare_pr_preview: No such file or directory
Error: Process completed with exit code 1.
```

--- a/lib/find-current-git-tag.sh
+++ b/lib/find-current-git-tag.sh
@@ -24,11 +24,11 @@ fi
 echo >&2 "Determining Git tag for $github_repository/$git_ref"
 
 echo >&2 "Cloning repository $github_repository at ref $git_ref"
-git clone --bare --single-branch --branch "$git_ref" "https://github.com/$github_repository" bare_pr_preview
+git clone --bare  "https://github.com/$github_repository" bare_pr_preview
 
 cd bare_pr_preview || exit 1
 
-action_version=$(git describe --tags --match "v*.*.*" || git describe --tags || git rev-parse HEAD)
+action_version=$(git describe --tags --match "v*.*.*" "$git_ref" || git describe --tags "$git_ref" || git rev-parse HEAD)
 
 echo "$action_version"
 exit 0

